### PR TITLE
Addeed :Playlist 

### PR DESCRIPTION
**Pull request by aiman-ansari**
Added playlist page.

- User can add playlist.
- User can add video to playlist.
- User can delete playlist.
- User can delete video from video.

### CODE REVIEW

File: src/Components/Playlist/PlaylistContainer.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            setDescription('')
+            setTitle('')
+        }
+        
+    }
+    return(
+        <div class="playlist-modal">
+            <div class="modal-body">
+                <div className="modal-header">
+                    {playlist!==undefined && playlist.length>0 ?
```
Comment (rahulyadav139) [inline src/Components/Playlist/PlaylistContainer.js]: Suggestion: you could have used optional chaining here.

File: src/Components/Playlist/PlaylistContainer.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    :
+                    <span className="mb-1">Create new Playlist</span>
+                    }
+                    <span><i className="bi bi-x" onClick={() => {
+                        setPlaylistModal(false)}}></i></span>
+                </div>
+                <div className="playlist-title">
+                    {playlist&&
+                        playlist.map((item) => 
+                            <div className="playlist-content">
```
Comment (rahulyadav139) [inline src/Components/Playlist/PlaylistContainer.js]: Suggestion: forgot to add key attribute here.

File: src/Components/Playlist/PlaylistContainer.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                onChange={(e) =>setTitle(e.target.value)}>
+            </input>
+            <input 
+                placeholder="Enter description"
+                className="description"
+                value={description}
+                onChange={(e) =>setDescription(e.target.value)}>
+            </input>
+            <div>
+            <div className="text-danger">{isError}</div>
```
Comment (rahulyadav139) [inline src/Components/Playlist/PlaylistContainer.js]: Suggestion: you 'isState' for boolean value like 'isError', 'isClosed' etc. I don't think 'isError' here is a boolean value.

File: src/Context/IconContainerContext.js [hunk 4]
```diff
@@ -6,8 +6,10 @@ const useIconContainer = () => useContext(IconContainerContext)
 const IconContainerProvider = ({children}) =>{
     const [iconContainer, setIconContainer] = useState(null)
     const [show, setShow] = useState(false)   
+    const [iconContainer1, setIconContainer1] = useState(null)
+    const [show1, setShow1] = useState(false)
```
Comment (rahulyadav139) [inline src/Context/IconContainerContext.js]: Suggestion: naming convention for states could have been better.

File: src/Context/PlaylistContext.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import { createContext, useContext, useState, useEffect } from "react";
+import { toast } from "react-toastify";
+const PlaylistContext = createContext()
+const usePlaylist = () => useContext(PlaylistContext)
+const PlaylistContextProvider = ({children}) =>{
+    const [playlist, setPlaylist] = useState([])
+    const [playlistModal, setPlaylistModal] = useState(false)
+    useEffect(()=>{
+        getPlaylist()
+    },[])
```
Comment (rahulyadav139) [inline src/Context/PlaylistContext.js:11]: Suggestion: never leave a dependency array empty if there is some external variable/function/object there in useEffect.

File: src/Pages/Playlist/Playlist.js [hunk 6]
```diff
@@ -1,7 +1,58 @@
...
     return(
-        <>
-            Playlist page
-        </>
+        <div className='playlist-container'>
+            {isLogin ? 
+                playlist.length>0 ? <>
+                {
+                    playlist.map((item) => 
+                        <div className='playlist'>
```
Comment (rahulyadav139) [inline src/Pages/Playlist/Playlist.js]: Suggestion: use key attribute whenever you are using map method for rendering.

**Review by rahulyadav139 — APPROVED** [review]

File: src/Pages/Playlist/Playlist.js (hunk 6 above)
Comment (aiman-ansari) [reply src/Pages/Playlist/Playlist.js]: Sure I forgot it every single time

File: src/Components/Icons/IconContainer.js [hunk 7]
```diff
@@ -21,8 +26,28 @@ export const IconContainer = ({video}) =>{
...
+
+                    {playlistModal && <PlaylistContainer video={video}/>}  
                         <span>Save</span>
+                        </>
+                        // :
+                        // <>
+                        // <i class="bi bi-list-ul"
+                        //     onClick={() => {
+                        //         navigate('/login')
+                        // }}>
```
Comment (varnita619) [inline src/Components/Icons/IconContainer.js]: remove comments

**Review by varnita619 — APPROVED** [review]
Rest all is fine

**Comment by aiman-ansari** [comment]
Thank you for reviews did all changes as you guys suggested.

**Pull request merged**